shared/video-sdk/develop/live-streaming-over-multiple-channels/project-implementation/react-native.mdx

```diff
@@ -0,0 +1,240 @@
+    `onChannelMediaRelayStateChanged` event handler. In `App.tsx`, add the following method after `onUserOffline: (_connection, remoteUid) => {}`:
+
+    ``` ts
+    onChannelMediaRelayStateChanged: (state, code) => {
+        // This example shows toast messages when the relay state changes,
+        // a production level app needs to handle state change properly.
+        switch (state) {
+        case ChannelMediaRelayState.RelayStateConnecting: // RELAY_STATE_CONNECTING:
+            console.log('Channel media relay connecting.');
+            break;
+        case ChannelMediaRelayState.RelayStateRunning: // RELAY_STATE_RUNNING:
+            mediaRelaying = true;
+            console.log('Channel media relay running.');
+            break;
+        case ChannelMediaRelayState.RelayStateFailure: // RELAY_STATE_FAILURE:
+            mediaRelaying = false;
+            console.log('Channel media relay failure. Error code: ' + code);
+        }
+    },
+    ```
+
+3.  **Monitor channel media relay events**
+
+    To receive notifications of important channel relay events such as network disconnection, reconnection, and users joining channels, you add the `onChannelMediaRelayEvent` callback to `EventHandles`. In `App.tsx`, add the following method after `onChannelMediaRelayStateChanged: (state, code) => {}`:
+
+    ``` ts
+    onChannelMediaRelayEvent: code => {
+        switch (code) {
+            case ChannelMediaRelayEvent.RelayEventNetworkDisconnected: // RELAY_EVENT_NETWORK_DISCONNECTED
+                console.log(
+                'User disconnected from the server due to a poor network connection.',
+                );
+                break;
+            case ChannelMediaRelayEvent.RelayEventNetworkConnected: // RELAY_EVENT_NETWORK_CONNECTED
+                console.log('Network reconnected');
+                break;
+            case ChannelMediaRelayEvent.RelayEventPacketJoinedSrcChannel: // RELAY_EVENT_PACKET_JOINED_SRC_CHANNEL
+                console.log('User joined the source channel');
+        }
+    },
+    ```
+
+### Join multiple channels
+
+The alternate approach to multi-channel live streaming is joining multiple channels. In this section, you learn how to implement joining a second channel in your <Vpl k="CLIENT" />.
+
+#### Implement the user interface
+
+In this example, you need a button to join and leave a second channel. In `<View style={styles.btnContainer}>`, add the following code after `<Text onPress={leave} style={styles.button}>Leave</Text>`:
+``` html
+<Text onPress={multipleChannels} style={styles.button}>
+    Join Second Channel
+</Text>
+```
+
+#### Handle the system logic
+
+In your project, import the relevant libraries and declare the required variables.
+
+1.  **Add the required libraries**
+
+    To implement media relay, import the corresponding <Vg k="COMPANY" /> modules. In `App.tsx`, add the following imports before `createAgoraRtcEngine,`:
```

Comment:
Always explain things in a chronological order. 

To implement media relay, in`App.tsx`,:
      1. Replace the `IRtcEngine` library with `IRtcEngineEx`.
      2. Add the following imports before `createAgoraRtcEngine,`:

             ```
               code
              ```